chapter07/dtj/ex4

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/perl
+
+use warnings;
+use strict;
+
+print "Tell me about fred and his friends:\n";
+
+my @input = <>;
+
+foreach my $line ( @input ){
+  print $line if( $line =~ /([a-z]+[A-Z]+|[A-Z]+[a-z]+)\s*/);
```

Comment:
Not quite sure what the +'s and \s* will help you with as you don't have any bookends to your regex so they should have no effect.